### PR TITLE
(feat) 003: React module content

### PR DESCRIPTION
**Pull request by Dzmitry-Klokau**

### CODE REVIEW

**Review by Liza-Veis — APPROVED** [review]

**Review by annakosakova — APPROVED** [review]

**Review by Karina-Trosko — APPROVED** [review]

File: react/index.md [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+- - Components and props
+- React lifecycle​
+- React hooks and HOCs​
+- - Motivation
+- - Basic hooks
+- - Hook rules
+- - Custom hooks
+- - Hoc
+- Controlled/Uncontrolled components
+- Reconsilation
```
Comment (unfalse) [inline react/index.md:15]: **Reconciliation**

File: react/index.md [hunk 2]
```diff
@@ -1,7 +1,16 @@
 # Module content
 
-- ToDo
-- ToDo
-- ToDo
-- ToDo
-- ToDo
+- React foundation
+- - What is it React?​
```
Comment (unfalse) [inline react/index.md:4]: It looks weird. Just "What is React?"

**Comment by unfalse** [comment]
How about adding other features like context, portal, refs etc.?

**Comment by Liza-Veis** [comment]
> How about adding other features like context, portal, refs etc.?

@unfalse  I think we don't need to cover such topics in basic react module, we can leave links to resources in materials

**Pull request merged**